### PR TITLE
[MCH-147-46] feat: multicall scripts

### PR DESCRIPTION
**Pull request by skeletor-spaceman**

### CODE REVIEW

File: packages/yswaps/scripts/Router.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import { PendingTrade, TradeSetup } from './types';
+import uniswap from '@libraries/uniswap-v2';
+import { UNISWAP_V2_FACTORY, UNISWAP_V2_ROUTER, WETH } from '@deploy/mainnet-swappers/uniswap_v2';
+import { ethers, getChainId } from 'hardhat';
+import zrx from './libraries/zrx';
+
+export class Router {
```
Comment (skeletor-spaceman) [inline packages/yswaps/scripts/Router.ts:7]: This should be a library, not a class

File: packages/yswaps/scripts/execute-mainnet-pending-trades.ts [hunk 2]
```diff
@@ -28,92 +28,67 @@ let flashbotsProvider: FlashbotsBundleProvider;
...
     ethers.provider, // a normal ethers.js provider, to perform gas estimiations and nonce lookups
     signer // ethers.js signer wallet, only for signing request payloads, not transactions
   );
 
-  const tradeFactory = await ethers.getContract<TradeFactory>('TradeFactory');
+  // env setup for testing. Should not be necessary
+  const strategy: Signer = await impersonate('0xa48c616144FD4429b216A86388CAb0Eed990cE87');
+  const ymech: Signer = await impersonate('0x2C01B4AD51a67E2d8F02208F54dF9aC4c0B778B6');
+
+  let tf: TradeFactoryExecutor = TradeFactoryExecutor__factory.connect('0xBf26Ff7C7367ee7075443c4F95dEeeE77432614d', ymech);
```
Comment (skeletor-spaceman) [inline packages/yswaps/scripts/execute-mainnet-pending-trades.ts:48]: Quite weird to be using `X__factory`, we need to find a cleaner way.

Comment (skeletor-spaceman) [inline packages/yswaps/scripts/execute-mainnet-pending-trades.ts:48]:
```suggestion
  let tradeFactory: TradeFactoryExecutor = TradeFactoryExecutor__factory.connect('0xBf26Ff7C7367ee7075443c4F95dEeeE77432614d', ymech);
```
please use descriptive names

File: packages/yswaps/scripts/multicall/ThreePoolCrvMulticall.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+// 2) [usdc|usdt|dai] => yvBOOST
+// 3) yvBOOST withdraw  => yveCRV
+
+export class ThreePoolCrvMulticall implements IMulticall {
+  private threeCrv: string = '0x6c3F90f043a72FA612cbac8115EE7e52BDe6E490';
+  private yveCrv: string = '0xc5bDdf9843308380375a611c18B50Fb9341f502A';
+  private yvBoost: string = '0x9d409a0A012CFbA9B15F6D4B36Ac57A46966Ab9a';
+
+  match(trade: PendingTrade) {
+    return trade._strategy == '0xa48c616144FD4429b216A86388CAb0Eed990cE87' && trade._tokenIn == this.threeCrv && trade._tokenOut == this.yveCrv;
```
Comment (skeletor-spaceman) [inline packages/yswaps/scripts/multicall/ThreePoolCrvMulticall.ts:19]: avoid using addresses, set them as constants on top of file.
same on :25 :26 & :29

File: packages/yswaps/scripts/Router.ts (hunk 1 above)
Comment (poolpitako) [reply packages/yswaps/scripts/Router.ts:7]: a library is an file with functions, right?

**Pull request merged**